tasks/main.yml

```diff
@@ -84,3 +86,14 @@
     when: ssl_certs_generate_dh_param
     tags: [ssl-certs,configuration]
 
+  - stat: path={{ ssl_certs_combined_path }}
+    register: stat_combined
+    tags: [ssl-certs,configuration]
+
+  - name: Combine key and cert
+    assemble:
+      dest: "{{ ssl_certs_combined_path }}"
+      src: "{{ ssl_certs_path }}"
+      regexp: "({{ ssl_certs_privkey_path | basename }}|{{ ssl_certs_cert_path | basename }})"
+    when: (not stat_combined.stat.exists) or generate_key_result|changed or generate_cert_result|changed or copy_result_private_key|changed or copy_result_ssl_certificate|changed
```

Comment:
@Sispheor  If we remove all the register added by the PR, this when and the stat task just bellow:
it's should work exactly the same  it's normally the way assemble works)
I will prefer a simplification in that case.
I am 95% sure, it's for that I asking a second point of view

Reply:
OK, I'll test it.